test/gtest/uct/test_peer_failure.cc

```diff
@@ -35,6 +35,64 @@ class test_uct_peer_failure : public uct_test {
 
     virtual void init();
 
+    void test_purge_failed_peer(bool fill_resources) {
+        set_am_handlers();
+
+        send_recv_am(0);
+        send_recv_am(1);
+
+        const ucs_time_t loop_end_limit = ucs::get_deadline();
+        const size_t num_pend_sends = 3ul;
+        size_t num_pend_added = 0;
+        uct_pending_req_t reqs[num_pend_sends];
+        {
+            scoped_log_handler slh(wrap_errors_logger);
+
+            if (!fill_resources) {
+                kill_receiver();
```

Comment:
what is the big difference in killing the receiver before ot after?

Reply:
this changes behavior
e.g. if killing receiver after doing AM short, TCP will be able adding requests to the pending queue